packages/host/host.js

```diff
@@ -51,15 +52,15 @@ class HostConnector extends libLink.WebSocketClientConnector {
 			plugins[pluginInfo.name] = pluginInfo.version;
 		}
 
-		this.sendHandshake("register_host", {
-			token: this.hostConfig.get("host.controller_token"),
-			agent: "Clusterio Host",
+		this.sendHandshake(new libData.MessageRegisterHost(new libData.RegisterHostData(
```

Comment:
In cases like these, I prefer the arguments to be formatted so you can see what gets created/passed like this - it makes things far more readable at a glance

`this.sendHandshake(
    new libData.MessageRegisterHost(
        new libData.RegisterHostData(
			this.hostConfig.get("host.controller_token"),
			"Clusterio Host",
			version,
			this.hostConfig.get("host.name"),
			this.hostConfig.get("host.id"),
			this.hostConfig.get("host.public_address"),
			plugins,
		)
    )
);`

Reply:
I agree these nested class construction are not very readable, so I've changed the ones I could find with grep.